float32 act_vx
float32 act_vy